SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -271,7 +272,16 @@ - (instancetype)initOutgoingMessageWithTimestamp:(uint64_t)timestamp
     // New outgoing messages should immediately determine their
     // recipient list from current thread state.
     NSMutableDictionary<NSString *, TSOutgoingMessageRecipientState *> *recipientStateMap = [NSMutableDictionary new];
-    NSArray<NSString *> *recipientIds = [thread recipientIdentifiers];
+    NSArray<NSString *> *recipientIds;
+    if ([self isKindOfClass:[OWSOutgoingSyncMessage class]]) {
+        NSString *_Nullable localNumber = [TSAccountManager localNumber];
+        OWSAssert(localNumber);
+        recipientIds = @[
+            localNumber,
```

Comment:
We used to special-case "outgoing sync messages" in the message sender, but hopefully with this change we can do that less.